src/main/java/com/minecolonies/coremod/colony/ColonyView.java

```diff
@@ -86,10 +97,80 @@ public static void serializeNetworkData(@NotNull final Colony colony, @NotNull f
         buf.writeBoolean(colony.isManualHiring());
         //  Citizenry
         buf.writeInt(colony.getMaxCitizens());
+
+        final Set<Block> freeBlocks = colony.getFreeBlocks();
+        final Set<BlockPos> freePos = colony.getFreePositions();
+
+        buf.writeInt(freeBlocks.size());
+        for(final Block block : freeBlocks)
+        {
+            ByteBufUtils.writeUTF8String(buf, block.getRegistryName().toString());
+        }
+
+        buf.writeInt(freePos.size());
+        for(final BlockPos block : freePos)
+        {
+            BlockPosUtil.writeToByteBuf(buf, block);
+        }
+
         //  Citizens are sent as a separate packet
     }
 
     /**
+     * Get a copy of the freePositions list.
+     * @return the list of free to interact positions.
+     */
+    public List<BlockPos> getFreePositions()
+    {
+        return new ArrayList<>(freePositions);
+    }
+
+    /**
+     * Get a copy of the freeBlocks list.
+     * @return the list of free to interact blocks.
+     */
+    public List<Block> getFreeBlocks()
+    {
+        return new ArrayList<>(freeBlocks);
+    }
+
+    /**
+     * Add a new free to interact position.
+     * @param pos position to add.
+     */
+    public void addFreePosition(@NotNull final BlockPos pos)
+    {
+        freePositions.add(pos);
```

Comment:
This should not happen on the client side.
Packet should be send to the server. Update of the view!